src/boot/TranslationProvider.js

```diff
@@ -54,7 +54,7 @@ const makeGetText = (intl: IntlShape): GetText => {
  * vs. https://reactjs.org/docs/legacy-context.html .
  */
 class TranslationContextTranslator extends PureComponent<{
-  children: React$Node,
+  +children: React$Node,
```

Comment:
nit: the commit message mentions "nonempty exact types defined inline" but this one is inexact. Perhaps s/exact/object/ in that line?

... Also I don't see a reason why this *shouldn't* be exact. Probably good to make it so, either before or after this commit. Most likely it just got missed when I swept through making props types exact, because it was defined inline.

Oh, I guess the message says "fall largely", so it's not untrue as is. Still, easy to make it complete so might as well. And following the links, I believe all the same facts are just as true of inexact object types. Ah in fact here's the docs for `$ReadOnly` itself:
https://flow.org/en/docs/types/utilities/#toc-readonly
So effectively it seems to be *defined*, at an interface/docs level, as just applying `+` to each key.

Reply:
Made the relevant change. Added that link to footnote 2.

Reply:
Looks good! Just one quibble remaining, in the commit message:

    The former have been manually wrapped with `$ReadOnly`. The latter
    have had their entries prefixed with `+`, which for exact types has
    essentially the same effect.

I think this could more simply say "[...] prefixed with `+`, which means the same thing." That is: I believe it's exactly the same effect, by definition, and independent of whether the object type is exact. (That's based on that linked bit of docs.)

Does that sound right to you?

Reply:
It sounds plausible, but not definite: I read your phrasing "means the same thing" as "is identical", and the docs merely state that the two are "equivalent". Equivalent modulo what?

(I'll punt on that if they will. Re-revised.)

Reply:
New text LGTM!

    The former have been manually wrapped with `$ReadOnly`. The latter
    have had their entries prefixed with `+`, which is functionally
    equivalent. (See, e.g., the Flow documentation [2].)

---

FWIW, I think by "equivalent" they mean that they're completely interchangeable for the question of whether a program is well-typed or not. IOW from a type-theory perspective they are different ways of spelling the exact same type; the type-checker might remember which way you spelled it, but that's only a concession to the needs of good error messages for humans.

That interpretation comes partly from scanning for other uses of "equivalent" on that page, and partly from knowing that the Flow people are pretty type-theory-oriented and applying my mental model of how people talk and write in academic PL/types communities.